meilisearch/tests/client/test_client_dumps.py

```diff
@@ -0,0 +1,43 @@
+import json
+import time
+import meilisearch
+from meilisearch.tests import BASE_URL, MASTER_KEY, clear_all_indexes
+
+class TestClientDumps:
+
+    """ TESTS: Client dumps creation and status """
+
+    client = meilisearch.Client(BASE_URL, MASTER_KEY)
+    index = None
+    dataset_file = None
+    dataset_json = None
+
+    def setup_class(self):
+        clear_all_indexes(self.client)
+        self.index = self.client.create_index(uid='indexUID')
+        self.dataset_file = open('./datasets/small_movies.json', 'r')
+        self.dataset_json = json.loads(self.dataset_file.read())
+        self.dataset_file.close()
+
+    def test_dump_creation(self):
+        """Tests the creation of a MeiliSearch dump"""
+        dump = self.client.create_dump()
+        assert dump['uid'] is not None
+        assert dump['status'] is not None
+        assert dump['status'] == 'processing'
+
+    def test_dump_status_route(self):
+        """Tests the route for getting a MeiliSearch dump status"""
+        response = self.index.add_documents(self.dataset_json, primary_key='id')
+        self.index.wait_for_pending_update(response['updateId'])
+        dump = self.client.create_dump()
+        assert dump['uid'] is not None
+        assert dump['status'] is not None
+        dump_status = self.client.get_dump_status(dump['uid'])
+        assert dump_status['uid'] is not None
+        assert dump_status['status'] is not None
+        assert dump_status['status'] == 'processing'
+        while dump_status['status'] == 'processing':
+            time.sleep(0.1)
+            dump_status = self.client.get_dump_status(dump['uid'])
+        assert dump_status['status'] == 'done'
```

Comment:
From my POV, you are testing MeiliSearch here. The SDK should not test the dump is created, in particular in `test_dump_status_route` that only returns the status of a dump.

Reply:
Indeed!

But at the same time it checks that with a dump status change on MS, client recieves the new information. Doesn't hurt to have the check for free :)

If you think this is absolutely out of scope, I'm not against removing it tho!

Reply:
This check is not "for free". It adds a small complexity in test readability. A small one, but if you add more, it's not small at all 🙂
Now MeiliSearch is greatly doing its own tests, we should try now to stay in our scope to keep our SDKs as clean as possible. A lot of our old tests should not be in our SDKs anymore because they are also testing MeiliSearch, but before we find time to clean them, I suggest we don't add any other out-of-scope tests.
Also, @bidoubiwa, what do you think?

Reply:
I understand your POV and mostly agree with it. But also, I would still prefer to wait until the dump is created (while loop) because if we add a new test below (as I just did) we can introduce a conflict (because you are not allowed to create another dump if one is being created). So it would be better to wait until the dump is done to quit the test. In this case, I would remove just the last line of this block, which is why I said it is 'free' to check the 'done' status, WDYT?

Reply:
I did abstract this waiting loop on a `wait_for_dump_creation` method just used in tests, to facilitate reusability and readability.

It now looks like:

```python
    assert dump_status['status'] == 'processing'
    wait_for_dump_creation(dump_status['uid'])
    assert dump_status['status'] == 'done'
```

If you want me to remove the last line, react with 👎 and I'll do ;)

Reply:
From my POV, I don't see the point to wait for the dump to be done, even in another test ^^ I referred to the whole block
That's not the role of the method. The method calls the route and returns the JSON as a dictionary with a `status` and a `uid` as keys. For me, no need to wait for the end of the dump creation. But if @bidoubiwa agrees with you, let's keep it!

Reply:
It is needed because if we exit the test right away, and a new test tries to create a new dump, it will throw an error, since you can't create two dumps at the same time

Reply:
In the other hand, I opted for removing the last check for the 'done' status as you suggested

Reply:
> It is needed because if we exit the test right away, and a new test tries to create a new dump, MeiliSearch will throw an error, since you can't create two dumps at the same time

I was slow to understand! sorryyyyy! 🐌 I've just understand that's only for the next tests! 😅

Reply:
But if it's for the next tests, why the test `test_dump_creation` does not have any `wait_for_pending_dump`, and most of all, does not need it?

Reply:
Because at the time that first test is executed, there is no index, no documents, MeiliSearch is completely empty, and therefore the dump creation is quite fast, almost instant. But you are rigth, in theory it should also wait for it's creation. In the other hand, for the get stattus test, MeiliSearch creates an index and populate it before creating a dump, and so it takes some time.